src/prettyprint.lua

```diff
@@ -0,0 +1,229 @@
+local COLOURS = require("src.utils.consoleColor")
+local PASS_SYMBOL = COLOURS.PASS("√")
+local ERROR_SYMBOL = "●"
+local FAIL_SYMBOL = COLOURS.FAIL("×")
+local PASS_HEADER = COLOURS.PASS_HEADER(" PASS ")
+local FAIL_HEADER = COLOURS.FAIL_HEADER(" FAIL ")
+
+-- Matches Jest
+local TAB_SIZE = 2
+
+local NodeType = require("src.interface.testnodetype")
+local tablex = require("src.utils.tablex")
+local prettyValue = require("src.utils.prettyValue")
+
+local function getTabs(amount)
+	return (" "):rep(amount * TAB_SIZE)
+end
+
+--- Recursively traverses the node tree while calling the supplied callback.
+---@param node lest.TestResult|lest.Describe
+---@param onNode fun(node: lest.TestResult|lest.Describe, depth: number):boolean A callback which is called for each Describe and Test in the tree. You can return true to keep going or return false to terminate.
+---@param onEnd fun()|nil A callback which is called once a branch in the tree has finished being traversed.
+local function traverseNodes(node, onNode, onEnd, _depth)
+	_depth = _depth or 0
+
+	for _, childNode in ipairs(node) do
+		-- If the callback does not want to traverse this specific branch anymore, it can request it by returning false.
```

Comment:
This is another "what", not a "why"

Reply:
This is still a "what" comment on the latest revision.

We probably don't need this anyway though, as the behaviour is documented in the annotations.

Reply:
Removed

Reply:
1. It has not been removed. If you removed it locally and didn’t push then it has not been removed in the context of this PR.
2. Do not resolve review threads created by other people.